evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -170,20 +171,33 @@ def _initialize_oversampler(self, X, y, sampler_class):
             y (pd.Series): Target features
             sampler_class (imblearn.BaseSampler): The sampler we want to initialize
         """
-        _, y_ww = self._prepare_data(X, y)
+        _, y_pd = self._prepare_data(X, y)
         sampler_params = {
             k: v
-            for k, v in copy.copy(self.parameters).items()
-            if k not in ["sampling_ratio", "sampling_ratio_dict"]
+            for k, v in self.parameters.items()
```

Comment:
Do you know why we copied before and why its okay that we remove this?

Reply:
This was something Karsten pointed out, but since we're making a copied dictionary from the parameters, I don't think it's necessary for us to copy the dictionary.

Reply:
We do a lot of copying throughout the code base....I feel like it *might* be someone (rightfully) scared of overwriting the originals.